centaur/src/main/resources/standardTestCases/three_step/three_step.cwl

```diff
@@ -0,0 +1,100 @@
+cwlVersion: v1.0
+class: Workflow
+id: three_step
+inputs:
+- id: pattern
+  type: string
+outputs:
+- id: cgrep-count
+  outputSource: "#cgrep/cgrep-count"
+  type: int
+- id: wc-count
+  outputSource: "#wc/wc-count"
+  type: int
+steps:
+- id: ps
+  in: []
+  out:
+  - id: ps-stdOut
+  run:
+    inputs: []
+    outputs:
+    - id: ps-stdOut
+      outputBinding:
+        glob: ps-stdOut.txt
+      type: File
+    class: CommandLineTool
+    baseCommand: ps
+    stdout: ps-stdOut.txt
+- id: cgrep
+  in:
+  - id: pattern
+    source: "#pattern"
+  - id: file
+    source: "#ps/ps-stdOut"
+  out:
+  - id: cgrep-count
+  run:
+    inputs:
+    - id: pattern
+      type: string
+    - id: file
+      type: File
+    outputs:
+    - id: cgrep-count
+      outputBinding:
+        glob: cgrep-stdOut.txt
+        loadContents: true
+        outputEval: $(parseInt(self[0].contents))
+      type: int
+    class: CommandLineTool
+    requirements:
+    - class: ShellCommandRequirement
+    - class: InlineJavascriptRequirement
+    arguments:
+    - valueFrom: grep
+      shellQuote: false
+    - valueFrom: $(inputs.pattern)
+      shellQuote: false
+    - valueFrom: ${return inputs.file}
+      shellQuote: false
+    - valueFrom: '|'
+      shellQuote: false
+    - valueFrom: wc
+      shellQuote: false
+    - valueFrom: -l
+      shellQuote: false
```

Comment:
is there really not a more concise expression for this in CWL? or is this the more common form that we want to focus on supporting?

Reply:
Afaik this is it but I have minimal experience in CWL writing